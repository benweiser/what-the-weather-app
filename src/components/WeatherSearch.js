import { PropTypes } from "prop-types";
import React from "react";
import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import { css } from "emotion";

const StyledTextField = css`
  width: 380px;
`;

class WeatherSearch extends React.PureComponent {
  state = {
    zipCode: ""
  };

  handleChange = e => {
    this.setState({ zipCode: e.target.value });
    e.preventDefault();
  };

  handleSubmit = e => {
    if (this.props.fetchWeather) {
      this.props.fetchWeather(this.state.zipCode);
      this.setState({ zipCode: "" });
    }
    e.preventDefault();
  };

  render() {
    return (
      <div className={this.props.className}>
        <form onSubmit={this.handleSubmit}>
          <TextField
            placeholder="Enter city name"
            className={StyledTextField}
            onChange={this.handleChange}
            label="Enter city name"
            value={this.state.value}
            aria-label="Enter city name"
          />
          <Button color="primary" variant="contained" type="submit">
            Submit Data
          </Button>
        </form>
      </div>
    );
  }
}

WeatherSearch.propTypes = {
  fetchWeather: PropTypes.func.isRequired
};

export default WeatherSearch;
