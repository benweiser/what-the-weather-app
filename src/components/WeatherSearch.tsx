import React, { useState } from 'react';
import Button from '@material-ui/core/Button';
import { css } from 'emotion';
import SearchOptions from './SearchOptions';
import WeatherSearchInput from './WeatherSearchInput';

export type SearchType = 'city' | 'zip' | 'coords';

export interface WeatherSearchProps {
  className?: string;
  onFetchWeather: (state: WeatherSearchState) => void;
  searchMethod?: SearchType;
}

export interface WeatherSearchState {
  value: string;
  searchMethod: SearchType;
}

const StyledSearchFormDiv = css`
  display: flex;
`;

const WeatherSearch = (props: WeatherSearchProps) => {
  const { className } = props;
  const [state, setState] = useState<WeatherSearchState>({
    searchMethod: 'city',
    value: ''
  });
  const { searchMethod, value } = state;
  return (
    <div className={className}>
      <form
        onSubmit={(e: React.MouseEvent<any>) => {
          e.preventDefault();
          const { onFetchWeather } = props;
          if (onFetchWeather) {
            onFetchWeather(state);
          }
        }}
      >
        <SearchOptions
          onSearchTypeSelect={(type: SearchType) =>
            setState({
              ...state,
              searchMethod: type
            })
          }
          selectedOption={searchMethod as SearchType}
        />
        <div className={StyledSearchFormDiv}>
          <WeatherSearchInput
            onChange={(e: React.FormEvent<HTMLInputElement>) =>
              setState({
                ...state,
                value: e.currentTarget.value
              })
            }
            type={searchMethod}
            value={value}
          />
          <Button
            data-testid="search-button"
            color="primary"
            variant="contained"
            type="submit"
          >
            Get Weather
          </Button>
        </div>
      </form>
    </div>
  );
};

export default WeatherSearch;
