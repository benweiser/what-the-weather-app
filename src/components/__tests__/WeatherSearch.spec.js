import renderer from "react-test-renderer";
import React from "react";
import WeatherSearch from "../WeatherSearch";

describe("Weather Widget Snapshot", () => {
  test("renders", () => {
    const component = renderer.create(
      <WeatherSearch fetchWeather={jest.fn()} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
