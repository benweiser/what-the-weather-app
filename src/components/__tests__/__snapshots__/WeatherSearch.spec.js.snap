// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Widget Snapshot renders 1`] = `
<div
  className={undefined}
>
  <form
    onSubmit={[Function]}
  >
    <div
      aria-describedby={undefined}
      aria-label="Enter city name"
      className="MuiFormControl-root-1 css-e8diyq"
    >
      <label
        className="MuiFormLabel-root-10 MuiInputLabel-root-5 MuiInputLabel-formControl-6 MuiInputLabel-animated-9"
        data-shrink={false}
        htmlFor={undefined}
      >
        Enter city name
      </label>
      <div
        className="MuiInput-root-15 MuiInput-formControl-16 MuiInput-underline-19"
      >
        <input
          aria-invalid={false}
          aria-required={false}
          autoComplete={undefined}
          autoFocus={undefined}
          className="MuiInput-input-23"
          defaultValue={undefined}
          disabled={false}
          id={undefined}
          name={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={undefined}
          onKeyUp={undefined}
          placeholder="Enter city name"
          readOnly={undefined}
          required={undefined}
          rows={undefined}
          type="text"
          value={undefined}
        />
      </div>
    </div>
    <button
      className="MuiButtonBase-root-52 MuiButton-root-28 MuiButton-contained-37 MuiButton-containedPrimary-38 MuiButton-raised-40 MuiButton-raisedPrimary-41"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="submit"
    >
      <span
        className="MuiButton-label-29"
      >
        Submit Data
      </span>
      <span
        className="MuiTouchRipple-root-55"
      />
    </button>
  </form>
</div>
`;
